src/java/com/puppetlabs/http/client/impl/ClientMetricFilter.java

```diff
@@ -0,0 +1,64 @@
+package com.puppetlabs.http.client.impl;
+
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.Metric;
+import org.apache.commons.lang3.StringUtils;
+
+public class ClientMetricFilter {
+
+    static class ClientFilter implements MetricFilter {
+        public boolean matches (String name, Metric metric) {
+            return name.startsWith(Metrics.METRIC_NAMESPACE);
+        }
+    }
+
+    static class UrlFilter implements MetricFilter {
+        private String url;
+        private Metrics.MetricType metricType;
+
+        public UrlFilter(String url, Metrics.MetricType metricType) {
+            this.url = url;
+            this.metricType = metricType;
+        }
+
+        public boolean matches(String s, Metric metric) {
+            String[] metricNameArray = { Metrics.METRIC_NAMESPACE, Metrics.URL_NAMESPACE, url,
+                    Metrics.metricTypeString(metricType) };
+            return s.equals(StringUtils.join(metricNameArray, "."));
```

Comment:
For the places we join the raw names from the array to do the match, I wonder if it would be better to use the `MetricRegistry/name` helper method, like we do in `Metrics.startBytesReadUrlTimers`.  Seems like it would be best to use the same method everywhere that we join the segments of a name together?